src/nvim/api/vim.c

```diff
@@ -119,6 +119,28 @@ Integer nvim_get_hl_id_by_name(String name)
   return syn_check_group(name.data, name.size);
 }
 
+/// Gets all highlight groups
+///
+/// @note Unlike the `:highlight` command which list all the current highlight
+///       groups that have attributes set in human readable format, this function
+///       return a table of all current highlight groups in a machine readable format
+///       (i.e. can be directly fed to nvim_set_hl)
+///
+/// @param ns_id Get all highlight groups using namespace ns_id |nvim_get_namespaces()|.
+///              Use 0 to get all global highlight groups |:highlight|.
+/// @param val  Get all highlight groups options
+///                 - name: string
+///                 - id: integer
+///                 - rgb: (boolean, default false) shows colors in RGB hex representation instead of decimal.
```

Comment:
let's call it `hex`. it is a rgb value regardless (and in other contexts "rgb" means rgb instead of index color)

Reply:
`bg = "Black"` is also valid, do we want to add this option?

Reply:
valid for _setting_ but is it also returned as such?

Reply:
(nope, so this is irrelevant here ;) `hex` is better)

Reply:
I'm getting `bg = "#cdd6f4"` after setting `guibg=Black` instead of `bg = "#000000"`. Not sure why:

```C
char bg_hex[8];
coloridx_to_name(sgp->sg_rgb_bg_idx, sgp->sg_rgb_bg, bg_hex);
PUT_C(hl, "bg", STRING_OBJ(cstr_as_string(strdup(bg_hex))));
```

```
:lua=vim.api.nvim_get_hl(0, { name = "Normal" })
{
  bg = "#000000",
  fg = "#cdd6f4"
}
:hi Normal guibg=Black
:lua=vim.api.nvim_get_hl(0, { name = "Normal" })
{
  bg = "#cdd6f4",
  fg = "#cdd6f4"
}
```

Reply:
Works for me (but don't test with `Normal`, that group is not like the other kids...)

Reply:
Even with `String`

```lua
:hi String guibg=Black
:lua=vim.api.nvim_get_hl(0, { name = "String" })
{
  bg = "#a6e3a1",
  fg = "#a6e3a1"
}
```

bg's hex is whatever fg value might be

Reply:
Ah, then there's a bug in your code ;) (Make sure to compare with the output of the old `nvim_get_hl`.)

Reply:
Yes, what I don't understand is `highlight_list_one` also called `coloridx_to_name(sgp->sg_rgb_bg_idx, sgp->sg_rgb_bg, bg_hex)` but can perfectly handle `guibg=Black` argument

I will look more into it, but I think it has something to do with the `name_to_color` function

Reply:
> Make sure to compare with the output of the old `nvim_get_hl`.

The old nvim_get_hl doesn't have `hex` option so it just outputs `background = 0`

Reply:
Ok so the vim's function also has this problem

```
:hi String 
String         xxx guifg=#a6e3a1 guibg=Black
Debug: guibg=#a6e3a1
```

Reply:
Resolved.